src/main/java/com/conveyal/r5/profile/PerTargetPropagater.java

```diff
@@ -0,0 +1,107 @@
+package com.conveyal.r5.profile;
+
+import com.conveyal.r5.streets.LinkedPointSet;
+import gnu.trove.map.TIntIntMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+
+/**
+ * This is similar to the Propagater class, but instead of the reducer function being called with each iteration,
```

Comment:
The reason I pass it as a function is so that the times can be dealt with and summarized in GridComputer; the array of times is too large to hold in memory all at once, so it needs to be dealt with piecemeal. A member function of PerTargetPropagater would not have access to the variables in GridComputer (unless we made them public or package-private class fields and passed in the GridComputer instance, but I'm not sure that's better).